### PR TITLE
 doc: Requirements for the eviction

### PR DESCRIPTION
**Pull request by mmichal10**
Signed-off-by: Michal Mielewczyk <michal.mielewczyk@intel.com>

### CODE REVIEW

File: doc/requirements/eviction [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+group: Eviction
+---
+
+Eviction controlls a process of invalidating clean data from cache.
```
Comment (arutk) [inline doc/requirements/eviction:5]: There are other ways to invalidate cachelines as well. I would say that "Eviction is a process of invalidating cachelines in order to map to a new addresses"

File: doc/requirements/eviction [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+id: priorities
+---
+
+The product shall shall perfrom eviction based on the priority information
+associated with IO classes.
+
+The product shall evict in following order:
+
+1. IO classes with excedeed occupancy limit.
+2. Lower priority IO classes (omit pinned IO Classes)
```
Comment (arutk) [inline doc/requirements/eviction:19]: Like mentioned in other review: is "pinning" a term that is described anywhere in user documentation?

**Review by arutk — APPROVED** [review]

File: doc/requirements/eviction [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Eviction controlls a process of invalidating clean data from cache.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Eviction priorities
+id: priorities
+---
+
+The product shall shall perfrom eviction based on the priority information
```
Comment (Deixx) [inline doc/requirements/eviction:13]:
```suggestion
The product shall perform eviction based on the priority information
```

File: doc/requirements/eviction [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+associated with IO classes.
+
+The product shall evict in following order:
```
Comment (Deixx) [inline doc/requirements/eviction:16]:
```suggestion
The product shall evict in the following order:
```

File: doc/requirements/eviction [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+1. IO classes with excedeed occupancy limit.
+2. Lower priority IO classes (omit pinned IO Classes)
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Cold data eviction
+id: cold
+---
+
+The product shall evict the data within a single IO class in last recently used order.
```
Comment (Deixx) [inline doc/requirements/eviction:27]: Adding the abbreviation is just a suggestion. You can skip that ;) Main change is last -> least ;)
```suggestion
The product shall evict the data within a single IO class in the least recently used (LRU) order.
```

**Review by Deixx — CHANGES_REQUESTED** [review]

File: doc/requirements/eviction (hunk 2 above)
Comment (mmichal10) [reply doc/requirements/eviction:19]: I proposed definition of pinned ioclass in ioclass requirements.

File: doc/requirements/eviction (hunk 5 above)
Comment (Ostrokrzew) [reply doc/requirements/eviction]: Typo still exists.

File: doc/requirements/eviction [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+1. IO classes with excedeed occupancy limit.
+2. Lower priority IO classes (omit pinned IO Classes)
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Cold data eviction
+id: cold
+---
+
+The product shall evict the data within a single IO class in the last recently
```
Comment (Ostrokrzew) [inline doc/requirements/eviction:27]: Typo still exists.

**Review by Ostrokrzew — APPROVED** [review]

File: doc/requirements/eviction [hunk 7]
```diff
@@ -0,0 +1,28 @@
+---
+group: Eviction
+---
+
+Eviction is a process of invalidating cachelines in order to map to a new addresses
```
Comment (Deixx) [inline doc/requirements/eviction:5]:
```suggestion
Eviction is a process of invalidating cache lines in order to map them to new addresses
```

File: doc/requirements/eviction [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Eviction priorities
+id: priorities
+---
+
+The product shall perfrom eviction based on the priority information
```
Comment (Deixx) [inline doc/requirements/eviction:13]:
```suggestion
The product shall perform eviction based on the priority information
```

File: doc/requirements/eviction [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+associated with IO classes.
+
+The product shall evict in the following order:
```
Comment (Deixx) [inline doc/requirements/eviction:16]:
```suggestion
The product shall evict data from IO classes ordered as follows:
```

File: doc/requirements/eviction [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+1. IO classes with excedeed occupancy limit.
+2. Lower priority IO classes (omit pinned IO Classes)
```
Comment (Deixx) [inline doc/requirements/eviction:19]: specify if higher number means higher or lower priority

File: doc/requirements/eviction [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Cold data eviction
+id: cold
+---
+
+The product shall evict the data within a single IO class in the least recently
```
Comment (Deixx) [inline doc/requirements/eviction:27]:
```suggestion
The product shall evict data within a single IO class in the least recently
```

**Review by Deixx — CHANGES_REQUESTED** [review]

File: doc/requirements/eviction (hunk 10 above)
Comment (mmichal10) [reply doc/requirements/eviction:19]: I would say it's an implementation detail.

Comment (Deixx) [reply doc/requirements/eviction:20]: agreed

**Review by Deixx — APPROVED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Pull request merged**